src/pages/index.js

```diff
@@ -0,0 +1,299 @@
+    })
+    .catch((err) => console.error("Error updating avatar:", err))
+    .finally(() => {
+      setButtonText(dom.avatarEditSaveBtn, "Save");
+      disableBtn(dom.avatarEditSaveBtn, settings);
+    });
+}
+dom.avatarEditForm.addEventListener("submit", handleAvatarEditSave);
+
+addOpenModalListener(dom.avatarEditBtn, dom.avatarEditModal, () => {
+  resetValidation(
+    dom.avatarEditModal,
+    Array.from(dom.avatarEditModal.querySelectorAll(settings.inputSelector)),
+    settings
+  );
+});
+addCloseModalListener(dom.avatarCloseBtn, dom.avatarEditModal);
+
+addOpenModalListener(dom.profileEditBtn, dom.profileEditModal, () => {
+  dom.editProfileName.value = dom.profileName.textContent;
+  dom.editProfileDescription.value = dom.profileDescription.textContent;
+  resetValidation(
+    dom.editProfileForm,
+    Array.from(dom.editProfileForm.querySelectorAll(settings.inputSelector)),
+    settings
+  );
+});
+addCloseModalListener(dom.closeEditModalBtn, dom.profileEditModal);
+
+addOpenModalListener(dom.profileAddBtn, dom.profileAddModal, () => {
+  dom.profileAddForm.reset();
+  resetValidation(
+    dom.profileAddForm,
+    Array.from(dom.profileAddForm.querySelectorAll(settings.inputSelector)),
+    settings
+  );
+  disableBtn(
+    dom.profileAddForm.querySelector(settings.submitButtonSelector),
+    settings
+  );
+});
+addCloseModalListener(dom.closeAddModalBtn, dom.profileAddModal);
+
+addCloseModalListener(dom.closeDeletePostModalBtn, dom.deletePostModal);
+addCloseModalListener(dom.cancelDeletePostBtn, dom.deletePostModal);
+addCloseModalListener(dom.closePreviewModalBtn, dom.previewModal);
+
+dom.editProfileForm.addEventListener("submit", (evt) => {
+  evt.preventDefault();
+  setButtonText(dom.modalSaveBtn, "Saving...");
+  api
+    .editUserInfo({
+      name: dom.editProfileName.value,
+      about: dom.editProfileDescription.value,
+    })
+    .then((user) => {
+      dom.profileImg.src = user.avatar;
+      dom.profileName.textContent = user.name;
+      dom.profileDescription.textContent = user.about;
+      closeModal(dom.profileEditModal);
+    })
+    .catch((err) => console.error("Error updating user info:", err))
+    .finally(() => {
+      setButtonText(dom.modalSaveBtn, "Save");
+      disableBtn(dom.modalSaveBtn, settings);
+    });
+});
+
+dom.profileAddForm.addEventListener("submit", (evt) => {
+  evt.preventDefault();
+  const addModalSaveBtn = dom.profileAddModal.querySelector(".modal__submit-btn");
+  addModalSaveBtn.textContent = "Saving...";
+  api
+    .addNewCard({
+      name: dom.profileAddCaption.value,
+      link: dom.profileAddLink.value,
+    })
+    .then((card) => {
+      renderCard(card, dom.cardsList);
+      dom.profileAddForm.reset();
+      disableBtn(dom.modalSaveBtn, settings);
+      closeModal(dom.profileAddModal);
+    })
+    .catch((err) => console.error("Error adding new card:", err))
+    .finally(() => {
+      dom.modalSaveBtn.textContent = "Save";
```

Comment:
` dom.modalSaveBtn.textContent = "Save";`


Please, use the correct submit button here. It should be `addModalSaveBtn`



Please, reuse the `setButtonText` function, to avoid code duplication

```js

    .finally(() => {
      setButtonText(addModalSaveBtn, "Save");
    });
```